src/SourceLocator/Type/SourceFilter/SourceFilter.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Roave\BetterReflection\SourceLocator\Type\SourceFilter;
+
+interface SourceFilter
+{
+    public function getKey(): string;
+
+    public function isAllowed(string $source, ?string $name, ?string $filename = null): bool;
+}
```

Comment:
I think this API is too specific to your use-case, to be considered as part of the library itself.

Also, following general FP patterns, and having BC boundaries in mind, I'd rather say that `isAllowed()` should be relying on a userland callback, and since each locator operates slightly differently, it's hard to design a generic `Context` object to be used for filtering decisions